app/controllers/subscriptions_controller.rb

```diff
@@ -5,19 +5,21 @@ class SubscriptionsController < ApplicationController
 
   def create
     @subscription = current_user.subscriptions.build(conference_id: @conference.id)
-    if @subscription.save!
-      redirect_to root_path, notice: "You have been subscribed to receive email notifications for #{@conference.short_title}."
+    if @subscription.save
+      redirect_to root_path, notice: "You have subscribed to receive email notifications for #{@conference.title}."
     else
-      redirect_to root_path, error: subscription.errors.full_messages.to_sentence
+      redirect_to root_path, error: @subscription.errors.full_messages.to_sentence
     end
   end
 
   def destroy
     @subscription = current_user.subscriptions.find_by(conference_id: @conference.id)
+
+    (redirect_to root_path, error: "You are not subscribed to #{@conference.title}.") && return unless @subscription
     if @subscription.destroy
-      redirect_to root_path, notice: "You have been unsubscribed and now you will not be receiving email notifications for #{@conference.short_title}."
+      redirect_to root_path, notice: "You have been unsubscribed and now you will not be receiving email notifications for #{@conference.title}."
     else
-      redirect_to root_path, error: @subscription.errors.full_messages.to_sentence
+      redirect_to root_path, error: "You are not subscribed to #{@conference.title}."
```

Comment:
def destroy:
  check if there is the subscription.
  if no: go back with an error
  if yes: destroy it
  //   question: Is there the possibility that this destroy() function may fail for other reason than the absence of subscription ?